src/Service/Service.php

```diff
@@ -0,0 +1,196 @@
+<?php
+
+namespace TheAentMachine\Service;
+
+use Opis\JsonSchema\ValidationError;
+use Opis\JsonSchema\Validator;
+use TheAentMachine\Service\Exception\ServiceException;
+
+class Service implements \JsonSerializable
+{
+    /** @var string */
+    protected $serviceName;
+    /** @var string */
+    protected $image;
+    /** @var int[]|string[] */
+    protected $internalPorts;
+    /** @var string[] */
+    protected $dependsOn;
+    /** @var mixed[] */
+    protected $ports;
+    /** @var mixed[] */
+    protected $labels;
+    /** @var mixed[] */
+    protected $environment;
+    /** @var mixed[] */
+    protected $volumes;
+    /** @var \stdClass */
+    protected $validatorSchema;
+
+    /**
+     * Service constructor.
+     */
+    public function __construct()
+    {
+        $this->serviceName = '';
+        $this->image = '';
+        $this->internalPorts = array();
```

Comment:
You can use short array syntax ([] intead of array()).
You can initialize empty arrays in the field directly: `protected $labels = [];`